File: dz-playlist/src/main/java/com/queryinterface/dz/playlist/kafka/messages/TrackSearchSucceeded.java
package com.queryinterface.dz.playlist.kafka.messages;

import java.util.List;

import com.queryinterface.dz.playlist.dto.SearchResultDto;

public record TrackSearchSucceeded(long playlistId, long trackId, List<SearchResultDto> results) implements KafkaMessage {
    
    @Override
    public String getType() {
        return "track.search.succeeded.v1";
    }
}

File: yt-scraper/Kafka/TrackSearchSucceeded.cs
namespace yt_scraper
{
    public class TrackSearchSucceeded : Message
    {
        public long PlaylistId { get; set; }
        public long TrackId { get; set; }
        public List<SearchResult> Results { get; set; }
        public string GetMessageType()
        {
            return "track.search.succeeded.v1";
        }
    }
}